src/Xrefcheck/Verify.hs

```diff
@@ -255,32 +284,87 @@ verifyReference
-        else VerifyProgress{ vrLocal = moveProgress vrLocal, .. }
-      , ()
-      )
-    return $ fmap (WithReferenceLoc fileWithReference ref) res
-  else return mempty
+    = retryVerification 0 $ do
+        let locType = locationType rLink
+        if shouldCheckLocType mode locType
+        then case locType of
+          LocalLoc    -> checkRef rAnchor fileWithReference
+          RelativeLoc -> checkRef rAnchor
+                        (takeDirectory fileWithReference
+                          </> toString (canonizeLocalRef rLink))
+          AbsoluteLoc -> checkRef rAnchor (root <> toString rLink)
+          ExternalLoc -> checkExternalResource config rLink
+          OtherLoc    -> verifying pass
+        else return mempty
   where
+    retryVerification
+      :: Int
+      -> IO (VerifyResult VerifyError)
+      -> IO (VerifyResult $ WithReferenceLoc VerifyError)
+    retryVerification r resIO = do
+      res@(VerifyResult ves) <- resIO
+
+      let toRetry = any isFixable ves && r < maxRetries
+          currentDelay = extractRetryAfterInfo res
+
+      let moveProgress = alterOverallProgress r
+                       . alterProgressErrors res r
+
+      posixTime' <- getPOSIXTime
+      let posixTime = posixTimeToTimeSecond posixTime'
+
+      atomicModifyIORef' progressRef $ \VerifyProgress{..} ->
+        ( if isExternal $ locationType rLink
+          then VerifyProgress{ vrExternal =
+            let vrExternalAdvanced = moveProgress vrExternal
+            in if toRetry
+               then let mTTC = fmap fst . pTaskTimestamp $ vrExternal
+                        maxTime = maybe currentDelay (currentDelay `max`) mTTC
+                    in if maxTime == currentDelay
+                       then setTaskTimestamp maxTime posixTime vrExternalAdvanced
+                       else vrExternalAdvanced
+               else vrExternalAdvanced, .. }
+          else VerifyProgress{ vrLocal = moveProgress vrLocal, .. }
+        , ()
+        )
+      if toRetry
+      then do
+        threadDelay currentDelay
+        retryVerification (r + 1) resIO
+      else return $ fmap (WithReferenceLoc fileWithReference ref) res
+
+    alterOverallProgress
+      :: (Num a, Show a)
+      => Int
+      -> Progress a
+      -> Progress a
+    alterOverallProgress retryNumber
+      | retryNumber > 0 = id
+      | otherwise = incProgress
+
+    alterProgressErrors
+      :: (Num a, Show a)
+      => VerifyResult VerifyError
+      -> Int
+      -> Progress a
+      -> Progress a
+    alterProgressErrors res@(VerifyResult ves) retryNumber
+      | retryNumber >= maxRetries && any isFixable ves ||
+        retryNumber > 0 && any isUnfixable ves = fixableToUnfixable
+      | retryNumber > 0 && verifyOk res = decProgressFixableErrors
+      | verifyOk res || retryNumber > 0 && any isFixable ves = id
+      | any isFixable ves = incProgressFixableErrors
+      | otherwise = incProgressUnfixableErrors
```

Comment:
Oh we also need to account for the case when `maxRetries == 0`, i.e. when retrying is disabled. 

This is not possible at the moment (because `maxRetries` is hardcoded), but it soon will.

Think this should be just one additional guard `| maxRetries == 0 =` at the top.